src/write_gcm.c

```diff
@@ -3121,11 +3209,13 @@ static void wg_json_Points(json_ctx_t *jc, const wg_payload_t *element) {
   const wg_payload_value_t *value = &element->values[0];
   assert(!strcmp(value->name, "value"));
 
+  int metric_value_type = wg_metric_value_type_from_wg_payload_t(element);
+
   wg_typed_value_t typed_value;
   // We don't care what the name of the type is.
   const char *data_source_type_static;
-  if (wg_typed_value_create_from_value_t_inline(&typed_value,
-            value->ds_type, value->val, &data_source_type_static) != 0) {
+  if (wg_typed_value_create_from_value_t_inline(&typed_value, value->ds_type, 
+      metric_value_type, value->val, &data_source_type_static) != 0) {
```

Comment:
In reformatting this, you lost the indent. You need 2 spaces beyond the start of the function name (see, e.g., line 3219 below)...

Reply:
Done.